usr/src/cmd/install-tools/usbgen

```diff
@@ -324,40 +326,64 @@ devs="$(lofiadm -la "${usb_file}")" || \
 # Set rdevs by replacing dsk with rdsk in devs
 #
 rdevs="${devs/dsk/rdsk}"
-# for mount
+# for system partition
 s0devs="${devs/p0/s0}"
-# for newfs and installboot
-rs0devs="${rdevs/p0/s0}"
-# for prtvtoc | fmthard
+s2devs="${devs/p0/s2}"
+# for newfs and installboot and prtvtoc | fmthard
 rs2devs="${rdevs/p0/s2}"
+rs0devs="${rdevs/p0/s0}"
+pdev="${devs/p0/}"
 
 #
-# create Solaris2 partition
+# create GPT partitioning with system, boot and root
 #
-fdisk -B "${rdevs}"
+zpool create -B -o bootsize=34M usbgen-$$ $pdev
+zpool destroy usbgen-$$
```

Comment:
what's the point of doing this?

Reply:
It will create the GPT on disk. Probably should use fdisk -E instead...  I think the initial idea was to re-use the partitions created by zpool, but then it appeared we have to have boot partition for ufs anyhow (need that with GPT).